ingestion/src/metadata/ingestion/source/sql_source.py

```diff
@@ -102,7 +102,19 @@ def get_connection_url(self):
         url += f"{self.host_port}"
         if self.database:
             url += f"/{self.database}"
-        logger.info(url)
+        if self.options:
+            url = self.get_options(options=self.options, url=url)
+
+        logger.debug(url)
+        return url
+
+    def get_options(self, options, url):
+        if options:
+            url += "?" if "?" not in url else ""
+            for key, value in options.items():
+                if url[-1] != "?":
+                    url += "&"
+                url += f"{key}={value}"
```

Comment:
Ditto as below for using [urlencode](https://docs.python.org/3/library/urllib.parse.html#urllib.parse.urlencode)